module/VuFind/src/VuFind/Db/Table/Broadcasts.php

```diff
@@ -0,0 +1,304 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Db_Table
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Johannes Schultze <schultze@effective-webwork.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Db\Table;
+
+use Exception;
+use Laminas\Db\Adapter\Adapter;
+use VuFind\Db\Row\RowGateway;
+use VuFind\Db\Table\PluginManager as PluginManager;
+
+use function in_array;
+
+/**
+ * Table Definition for broadcasts
+ *
+ * @category VuFind
+ * @package  Db_Table
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Johannes Schultze <schultze@effective-webwork.de>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class Broadcasts extends Gateway
+{
+    /**
+     * Notifications config
+     *
+     * @var mixed
+     */
+    protected $config;
+
+    /**
+     * Constructor
+     *
+     * @param Adapter       $adapter Database adapter
+     * @param PluginManager $tm      Table manager
+     * @param array         $cfg     Laminas configuration
+     * @param mixed         $config  Notifications config
+     * @param RowGateway    $rowObj  Row prototype object (null for default)
+     * @param string        $table   Name of database table to interface with
+     */
+    public function __construct(
+        Adapter $adapter,
+        PluginManager $tm,
+        $cfg,
+        $config,
+        ?RowGateway $rowObj = null,
+        $table = 'notifications_broadcasts'
+    ) {
+        $this->config = $config;
+        parent::__construct($adapter, $tm, $cfg, $rowObj, $table);
+    }
+
+    /**
+     * Insert a new broadcast into the database or update an existing one..
+     *
+     * @param array $data          Data to be written to the database
+     * @param array $broadcastData Data of an existing broadcast
+     * @param array $broadcast_id  Id of the broadcast to be edited
+     *
+     * @throws Exception
+     */
+    public function insertOrUpdateBroadcast($data, $broadcastData = null, $broadcast_id = null)
+    {
+        foreach ($this->config['Notifications']['languages'] as $language) {
+            if ($broadcastData['id_' . $language] == null) {
+                $broadcast = $this->createRow();
+            } else {
+                $broadcast = $this->getBroadcastById($broadcastData['id_' . $language]);
+            }
+
+            $broadcast->visibility = $data['visibility'];
+            $broadcast->visibility_global = $data['visibility_global'];
+            $broadcast->priority = $data['priority'];
```

Comment:
I get an exception here, no `priority` field.  I was able to get past it with
```
            $broadcast->priority = $data['priority'] ?? -1;
```
but there may be a better default.